yt_dlp/extractor/microsoftembed.py

```diff
@@ -63,3 +71,221 @@ def _real_extract(self, url):
             'subtitles': subtitles,
             'thumbnails': thumbnails,
         }
+
+
+class MicrosoftMediusBaseIE(InfoExtractor):
+    @staticmethod
+    def _sub_to_dict(subtitle_list):
+        subtitles = {}
+        for sub in subtitle_list:
+            subtitles.setdefault(sub.pop('tag', None) or 'unknown', []).append(sub)
+        return subtitles
+
+    def _extract_ism(self, ism_url, video_id):
+        formats = self._extract_ism_formats(ism_url, video_id)
+        for fmt in formats:
+            if fmt.get('language') == 'eng' or 'English' in fmt.get('format_id', ''):
+                fmt['language_preference'] = -1
+            else:
+                fmt['language_preference'] = -10
+        return formats
+
+
+class MicrosoftMediusIE(MicrosoftMediusBaseIE):
+    _VALID_URL = r'https?://medius\.microsoft\.com/[^?#]+/(?P<id>[0-9a-f\-]+)'
```

Comment:
I think we should not match so broadly. These are the three URL formats I was able to find:
```
https://medius.microsoft.com/Embed/Video?id=0dc69bda-079b-4070-a7db-a8da1a06a9c7
https://medius.microsoft.com/Embed/video-nc/fe823a91-959c-465b-96d4-8f4db624f72c
https://medius.microsoft.com/Embed/VideoDetails/78493569-9b3b-4a85-a409-ee76e789e25c
```
I would then also adjust the code to download the same URL format, but just set the UUID.
Please add the examples to the tests, at least as `only_matching` as well.

Reply:
https://github.com/yt-dlp/yt-dlp/pull/9177/commits/30936fc806a20f51adc5aa2cd3a2c133ed5951e4